src/shared/luaEnvManager/LuaEnvironmentManager.ts

```diff
@@ -0,0 +1,132 @@
+import { LuaEnvironment } from "@lib/lua";
+import { SpriteRendererHandle } from "@shared/canvas/RendererHandle";
+import { KeyHandler } from "@shared/gameEngine/KeyHandler";
+
+export interface EnvData {
+  code: string,
+  output: string
+}
+
+interface ConstructorProps {
+  envData: EnvData,
+  rendererHandle: SpriteRendererHandle,
+  keyHandler: KeyHandler,
+  /**
+   * this function is used to set the envData.output only
+   * @param output the output to set
+   */
+  setOutput: ((output: string) => void)
+}
+
+class LuaEnvironmentManager {
+  private _error_prefix = "Error: ";
```

Comment:
This should be `static const`, but this isn't really useful anyway in my opinion as this isn't some kind of string that changes the behaviour of the class' implementation.